lib/preview/previewUtils.js

```diff
@@ -95,14 +97,60 @@ const silenceConsoleWhile = async (act, ...args) => {
   return result;
 }
 
+const memoize = (func, cacheBustCallback) => {
+  const cache = {};
+  return async (...args) => {
+    const stringArgs = args.toString();
+    const storedResult = cache[stringArgs];
+    if (storedResult && (cacheBustCallback ? !cacheBustCallback() : true)) {
+      //console.log(`Cache hit ${func}`)
+      return storedResult;
+    }
+    //console.log(`Cache miss ${func}`)
+    const res = await func(...args);
+    cache[stringArgs] = res;
+    return res;
+  }
+}
+
+const hidePreviewInDest = (previewDest) => previewDest.split('/').slice(2).join('/');
```

Comment:
Simple helper to hide implementation from the frontend (`@preview/<uuid>/my/dest` -> `my/dest`)